### PR TITLE
Numpy upper limit: `<2.0.0`

### PR DESCRIPTION
**Pull request by LinasKo**
# Description

As of a few days ago, an error is raised when attempting to load `cv2`, as it does not support numpy `2.0.0`. This results in our CI tests failing and may have consequences for end-users.

Until all our dependencies expand support, let's pin numpy to `<2.0.0`.  

See more: https://github.com/opencv/opencv-python/issues/943#issuecomment-2171817940

## Type of change

Please delete options that are not relevant.

-   [x] Bug fix (non-breaking change which fixes an issue)

## How has this change been tested, please provide a testcase or example of how you tested the change?

1. `pytest` passes
2. `poetry install --with dev` works
3. Verified that changes to the lock file are minimal
4. importing cv2 and numpy work
5. Hopefully CI builds succeeds, unlike in https://github.com/roboflow/supervision/pull/1286

## Any specific deployment considerations

I have not done anything with respect to versioning. We should think about releasing this as `0.21.1`.

### CODE REVIEW

**Review by onuralpszr — APPROVED** [review]

**Comment by onuralpszr** [comment]
I am merging now for keep dev workable.

**Comment by LinasKo** [comment]
I think it's a good decision to merge quickly this time; thanks @onuralpszr!

Though for anything larger than this (even if urgent), let's add some time for a core team to get a second opinion 😉

**Comment by onuralpszr** [comment]
> I think it's a good decision to merge quickly this time; thanks @onuralpszr!
> 
> Though for anything larger (even if urgent), let's add some time for a core team to get a second opinion 😉 


Thanks, For this one. I had to make sure dev is alive and for actual solution we need to find abi friendly way handling it for 1.x and 2.x if possible

**Pull request merged**
